R/expct.R

```diff
@@ -122,11 +122,7 @@ expct <- function(
   
   # Check query
   if (is.null(query)) {
-    if (any(is.na(evidence))) {
-      query <- params$meta$variable
-    } else {
-      query <- setdiff(params$meta$variable, colnames(evidence))
-    }
+    query <- params$meta$variable
```

Comment:
So you would always calculate the expectation for all variables if query not given, including the variables in evidence?

Reply:
Isn't this potentially inefficient if there are lots of conditioning variables? Or is there some offsetting speedup that I'm missing here?